articles/ssh-in-nodejs/index.md

```diff
@@ -0,0 +1,149 @@
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+```
+
+The `readline` library allows users to provide input using the command line. We are going to take the VM host details such as hostname, username, and password as input from the user. This is required to establish a secure connection with the remote host. 
+
+```node
+// function to get input from user
+function get_input() {
+
+	rl.question("What is the host? ", function(host) {
+	    rl.question("What is the user? ", function(user) {
+	    	rl.question("What is the password? ", function(password) {	
+	    		
+	    		ssh(host, user ,password);
+	   			rl.close();
+	    	
+	    	});
+		});
+	});
+}
+```
+
+the `get_input()` function takes input from the user. The hostname, username, and password of the VM are stored in the `host`, `user`, and `password` variables and are passed to the ssh() function as parameters.
+
+```node
+// function to ssh into a remote host.
+function ssh(host, user ,password){
+
+	console.log('inside the system')
+	var SSH = require('simple-ssh');
+	var ssh = new SSH({
+	    host: host,
+	    user: user,
+	    pass: password
+	});
+	// execute the df -h command to find out disk utilization
+	ssh.exec('df -h', {
+	    out: function(stdout) {
+	    	// console.log(stdout);
+	        parse(stdout);
+	    }
+	}).start();
+}
+```
+
+The `ssh()` function takes the hostname, username, and password as the input and establishes an SSH connection with the VM using simple-ssh. The `exec()` function executes a Linux command on the remote host. In this case, we execute the `df -h` command that shows us the amount of disk space that is free on the host. 
+
+Once we get the output of the `df -h` command in the stdout variable, we pass it to the `parse()` function that parses the output and displays it in a presentable format. 
+
+![df command](/engineering-education/ssh-in-node/df.png)
+
+The output of the df command is hard to read and understand. Therefore, we need to parse it and extract the relevant data.
+
+```node
+// Threshold value to see if disk utilization is below or above the desired limit.
+const threshold = 90;
+// function to parse the raw data from the df -h command.
+function parse(data){
+	// parser to parse the data(separated by tabs and spaces)
+	var parser = require('node-csv-parse');
+	
+	var parsed = parser(data, {
+		delimiter: ' ',
+		trim: true
+	}).asRows();
+
+	var finalData = [];
+
+	// go through every row in the data and add another column for threshold.
+	// Also, check to see if CPU utilization percentage has exceeded threshhold or not.
+	for(var i = 0; i < parsed.length; i++){
+		var temp = parsed[i][0].split(' ');
+		
+		// remove all the empty spaces in the array
+		temp = temp.filter(function(str) {
+    		return /\S/.test(str);
+		});
+		
+		if(i == 0){
+			temp.pop();
+			temp.push('Threshhold exceeded?');
+		}
+		else{
+			if(parseInt(temp[4]) >= threshhold){
```

Comment:
"Threshold" is misspelled a couple of times in this article, and in this case, it causes an error